src/test/java/seedu/foodrem/model/item/NameContainsKeywordsPredicateTest.java

```diff
@@ -69,9 +69,9 @@ public void test_nameDoesNotContainKeywords_returnsFalse() {
         predicate = new NameContainsKeywordsPredicate(List.of("Carrots"));
         assertFalse(predicate.test(new ItemBuilder().withItemName("Potato Cucumber").build()));
 
-        // Keywords match quantity, bought date and expiry date, but does not match name
+        // Keywords match quantity, bought date and expiry date, and name as it is a substring
         predicate = new NameContainsKeywordsPredicate(Arrays.asList("Potato", "12345", "11-11-2022", "12-12-2022"));
-        assertFalse(predicate.test(new ItemBuilder()
+        assertTrue(predicate.test(new ItemBuilder()
```

Comment:
Does not belong in this method, but fix later.

```suggestion
        // TODO: Move this to the correct test method
        assertTrue(predicate.test(new ItemBuilder()
```